conans/test/integration/package_info_test.py

```diff
@@ -235,3 +237,169 @@ def package_info(self):
         client.run("create conanfile.py dep/1.0@us/ch", assert_error=True)
         self.assertIn("dep/1.0@us/ch package_info(): self.cpp_info.components cannot be used "
                       "with self.cpp_info configs (release/debug/...) at the same time", client.out)
+
+    def package_info_wrong_cpp_info_test(self):
+        conanfile = textwrap.dedent("""
+            import os
+            from conans import ConanFile
+            class Dep(ConanFile):
+                def package_info(self):
+                    self.cpp_info.name = "Boost"
+                    self.cpp_info.components["Accumulators"].includedirs = [os.path.join("boost", 
+                    "accumulators")]
+                    self.cpp_info.libs = ["hello"]
+        """)
+
+        client = TestClient()
+        client.save({"conanfile.py": conanfile})
+        client.run("export . name/1.0@us/ch")  # Does NOT fail on export
+        client.run("create . name/1.0@us/ch", assert_error=True)
+        self.assertIn("name/1.0@us/ch package_info(): self.cpp_info.components cannot be used with "
+                      "self.cpp_info global values at the same time", client.out)
+
+    def package_info_components_complete_test(self):
+        dep = textwrap.dedent("""
+            import os
+            from conans import ConanFile
+            class Dep(ConanFile):
+                exports_sources = "*"
+                def package(self):
+                    self.copy("*")
+                def package_info(self):
+                    self.cpp_info.name = "Galaxy"
+                    self.cpp_info.components["Starlight"].includedirs = [os.path.join("galaxy", "starlight")]
+                    self.cpp_info.components["Starlight"].libs = ["libstarlight"]
+                    self.cpp_info.components["Planet"].includedirs = [os.path.join("galaxy", "planet")]
+                    self.cpp_info.components["Planet"].libs = ["libplanet"]
+                    self.cpp_info.components["Planet"].requires = ["Starlight"]
+                    self.cpp_info.components["Launcher"].system_libs = ["ground"]
+                    self.cpp_info.components["ISS"].includedirs = [os.path.join("galaxy", "iss")]
+                    self.cpp_info.components["ISS"].libs = ["libiss"]
+                    self.cpp_info.components["ISS"].libdirs = ["iss_libs"]
+                    self.cpp_info.components["ISS"].system_libs = ["solar", "magnetism"]
+                    self.cpp_info.components["ISS"].requires = ["Starlight", "Launcher"]
```

Comment:
I feel like that if it works for one of the attributes of `cpp_info` it should work for all of them, so, if we are not testing all the attributes it would be enough to test one them